From 5aafc0cd976a7528dea80d61ac4c01eb65adfce6 Mon Sep 17 00:00:00 2001
From: Rafael Martins <rmartins@anaconda.com>
Date: Sat, 29 Jun 2024 06:16:25 +0200
Subject: [PATCH 3/3] cmake: fix path for windows try run tests

---
 cmake/QtToolHelpers.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/cmake/QtToolHelpers.cmake b/cmake/QtToolHelpers.cmake
index 7674e7487d..4ebc0afd20 100644
--- a/cmake/QtToolHelpers.cmake
+++ b/cmake/QtToolHelpers.cmake
@@ -278,7 +278,9 @@ echo. > ${target_bin_dir}/${target}_try_run_passed"
         DEPENDS
             ${target}
         COMMAND
-            ${CMAKE_COMMAND} -E env QT_COMMAND_LINE_PARSER_NO_GUI_MESSAGE_BOXES=1
+            ${CMAKE_COMMAND} -E env
+                QT_COMMAND_LINE_PARSER_NO_GUI_MESSAGE_BOXES=1
+                PATH=$ENV{SRC_DIR}\\build\\qtbase\\lib\\qt6\\bin;$ENV{PREFIX}\\Library\\bin;$ENV{PATH}
             ${try_run_scripts_path}
         COMMENT
             "Testing ${target} by trying to run it."
-- 
2.39.2 (Apple Git-143)

