From 8e90bc749b6f83cd049d25ca7ce82d3fd9b36ab9 Mon Sep 17 00:00:00 2001
From: Rafael Martins <rmartins@anaconda.com>
Date: Sat, 29 Jun 2024 06:16:25 +0200
Subject: [PATCH 3/3] cmake: fix path for windows try run tests

---
 cmake/QtToolHelpers.cmake | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/cmake/QtToolHelpers.cmake b/cmake/QtToolHelpers.cmake
index 7674e7487d..a9db15972e 100644
--- a/cmake/QtToolHelpers.cmake
+++ b/cmake/QtToolHelpers.cmake
@@ -258,6 +258,8 @@ function(_qt_internal_add_try_run_post_build target try_run_flags)
     qt_configure_file(OUTPUT "${try_run_scripts_path}"
         CONTENT "@echo off
 
+set "PATH=$ENV{SRC_DIR}\build\qtbase\lib\qt6\bin;$ENV{PREFIX}\Library\bin;%PATH%"
+
 ${target_out_dir}/${target}.exe ${try_run_flags} > nul 2>&1
 
 if \"%errorlevel%\" == \"-1073741515\" (
-- 
2.39.2 (Apple Git-143)

