From e76e691c87fc901cf1cbb96340fb79f41cb6cb4d Mon Sep 17 00:00:00 2001
From: Rafael Martins <rmartins@anaconda.com>
Date: Sat, 29 Jun 2024 06:16:25 +0200
Subject: [PATCH 3/3] do not link rcc to zstd on windows

---
 src/tools/rcc/CMakeLists.txt | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/tools/rcc/CMakeLists.txt b/src/tools/rcc/CMakeLists.txt
index 55d4de2e28..a83d06292c 100644
--- a/src/tools/rcc/CMakeLists.txt
+++ b/src/tools/rcc/CMakeLists.txt
@@ -26,7 +26,9 @@ qt_internal_return_unless_building_tools()
 ## Scopes:
 #####################################################################
 
-qt_internal_extend_target(${target_name} CONDITION QT_FEATURE_zstd
-    LIBRARIES
-        WrapZSTD::WrapZSTD
-)
+if(NOT WIN32)
+    qt_internal_extend_target(${target_name} CONDITION QT_FEATURE_zstd
+        LIBRARIES
+            WrapZSTD::WrapZSTD
+    )
+endif()
-- 
2.39.2 (Apple Git-143)

