From 2c615335bb02bd02a62cd314a14890c244a9dee2 Mon Sep 17 00:00:00 2001
From: Rafael Martins <rmartins@anaconda.com>
Date: Sat, 29 Jun 2024 06:16:25 +0200
Subject: [PATCH 3/3] cmake: do not run try_run tests during build time

---
 cmake/QtToolHelpers.cmake | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/cmake/QtToolHelpers.cmake b/cmake/QtToolHelpers.cmake
index 7674e7487d..4dca0314dd 100644
--- a/cmake/QtToolHelpers.cmake
+++ b/cmake/QtToolHelpers.cmake
@@ -237,7 +237,7 @@ function(qt_internal_add_tool target_name)
         if(NOT arg_TRY_RUN_FLAGS)
             set(arg_TRY_RUN_FLAGS "-v")
         endif()
-        _qt_internal_add_try_run_post_build("${target_name}" "${arg_TRY_RUN_FLAGS}")
+        message(WARNING "would run: _qt_internal_add_try_run_post_build(\"${target_name}\" \"${arg_TRY_RUN_FLAGS}\")")
     endif()
 
     qt_enable_separate_debug_info(${target_name} "${install_dir}" QT_EXECUTABLE)
@@ -278,7 +278,8 @@ echo. > ${target_bin_dir}/${target}_try_run_passed"
         DEPENDS
             ${target}
         COMMAND
-            ${CMAKE_COMMAND} -E env QT_COMMAND_LINE_PARSER_NO_GUI_MESSAGE_BOXES=1
+            ${CMAKE_COMMAND} -E env
+                QT_COMMAND_LINE_PARSER_NO_GUI_MESSAGE_BOXES=1
             ${try_run_scripts_path}
         COMMENT
             "Testing ${target} by trying to run it."
-- 
2.39.2 (Apple Git-143)

