From 2e912196832e9fcaa29e8ea28e76db6ca9cfa94a Mon Sep 17 00:00:00 2001
From: Andrii Osipov <aosipov@anaconda.com>
Date: Mon, 26 Jan 2026 02:55:44 -0800
Subject: [PATCH] Set XCode version 15

---
 cmake/QtPublicAppleHelpers.cmake | 35 ++++++++++++++++----------------
 1 file changed, 18 insertions(+), 17 deletions(-)

diff --git a/cmake/QtPublicAppleHelpers.cmake b/cmake/QtPublicAppleHelpers.cmake
index 3ffe53c6..ce4d9f52 100644
--- a/cmake/QtPublicAppleHelpers.cmake
+++ b/cmake/QtPublicAppleHelpers.cmake
@@ -929,23 +929,24 @@ function(_qt_internal_get_apple_sdk_version out_var)
 endfunction()
 
 function(_qt_internal_get_xcode_version_raw out_var)
-    set(xcode_version "")
-    if(APPLE)
-        _qt_internal_execute_xcrun(xcode_version
-            XCRUN_ARGS xcodebuild -version
-            OUT_ERROR_VAR xcrun_error
-        )
-
-        string(REPLACE "\n" " " xcode_version "${xcode_version}")
-        string(STRIP "${xcode_version}" xcode_version)
-
-        if(NOT xcode_version)
-            message(FATAL_ERROR
-                    "Can't determine Xcode version. Is Xcode installed?"
-                    " Error details:\n${xcrun_error}")
-        endif()
-    endif()
-    set(${out_var} "${xcode_version}" PARENT_SCOPE)
+#    set(xcode_version "")
+#    if(APPLE)
+#        _qt_internal_execute_xcrun(xcode_version
+#            XCRUN_ARGS xcodebuild -version
+#            OUT_ERROR_VAR xcrun_error
+#        )
+#
+#        string(REPLACE "\n" " " xcode_version "${xcode_version}")
+#        string(STRIP "${xcode_version}" xcode_version)
+#
+#        if(NOT xcode_version)
+#            message(FATAL_ERROR
+#                    "Can't determine Xcode version. Is Xcode installed?"
+#                    " Error details:\n${xcrun_error}")
+#        endif()
+#    endif()
+#    set(${out_var} "${xcode_version}" PARENT_SCOPE)
+    set(${out_var} "15.0" PARENT_SCOPE)
 endfunction()
 
 function(_qt_internal_get_xcode_version out_var)
-- 
2.39.3 (Apple Git-146)

