From a90870796f3f5a2be181ecbf547f0a4489eced3c Mon Sep 17 00:00:00 2001
From: Brian Wing <bwing@anaconda.com>
Date: Tue, 15 Jul 2025 19:19:36 -0400
Subject: [PATCH] Set XCode version to one matching the SDK and compiler in use.

---
 cmake/QtPublicAppleHelpers.cmake | 34 ++++++++++++++++----------------
 1 file changed, 17 insertions(+), 17 deletions(-)

diff --git a/cmake/QtPublicAppleHelpers.cmake b/cmake/QtPublicAppleHelpers.cmake
index 8192b355c9f..cc0f11522ce 100644
--- a/cmake/QtPublicAppleHelpers.cmake
+++ b/cmake/QtPublicAppleHelpers.cmake
@@ -898,23 +898,23 @@ function(_qt_internal_get_apple_sdk_version out_var)
 endfunction()
 
 function(_qt_internal_get_xcode_version_raw out_var)
-    set(xcode_version "")
-    if(APPLE)
-        _qt_internal_execute_xcrun(xcode_version
-            XCRUN_ARGS xcodebuild -version
-            OUT_ERROR_VAR xcrun_error
-        )
-
-        string(REPLACE "\n" " " xcode_version "${xcode_version}")
-        string(STRIP "${xcode_version}" xcode_version)
-
-        if(NOT xcode_version)
-            message(FATAL_ERROR
-                    "Can't determine Xcode version. Is Xcode installed?"
-                    " Error details:\n${xcrun_error}")
-        endif()
-    endif()
-    set(${out_var} "${xcode_version}" PARENT_SCOPE)
+#    set(xcode_version "")
+#    if(APPLE)
+#        _qt_internal_execute_xcrun(xcode_version
+#            XCRUN_ARGS xcodebuild -version
+#            OUT_ERROR_VAR xcrun_error
+#        )
+#
+#        string(REPLACE "\n" " " xcode_version "${xcode_version}")
+#        string(STRIP "${xcode_version}" xcode_version)
+#
+#        if(NOT xcode_version)
+#            message(FATAL_ERROR
+#                    "Can't determine Xcode version. Is Xcode installed?"
+#                    " Error details:\n${xcrun_error}")
+#        endif()
+#    endif()
+    set(${out_var} "15.0" PARENT_SCOPE)
 endfunction()
 
 function(_qt_internal_get_xcode_version out_var)
-- 
2.49.0

