From 6a9026445784dff0c68b22582ecae871ab6eef7a Mon Sep 17 00:00:00 2001
From: Andrii Osipov <aosipov@anaconda.com>
Date: Mon, 26 Jan 2026 03:12:23 -0800
Subject: [PATCH] Removes the macOS-specific CFBundleGetMainBundle

---
 src/corelib/global/qlibraryinfo.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/corelib/global/qlibraryinfo.cpp b/src/corelib/global/qlibraryinfo.cpp
index e8b954a7..4ab5d742 100644
--- a/src/corelib/global/qlibraryinfo.cpp
+++ b/src/corelib/global/qlibraryinfo.cpp
@@ -108,7 +108,7 @@ static std::unique_ptr<QSettings> findConfiguration()
     QString qtconfig = QStringLiteral(":/qt/etc/qt.conf");
     if (QResource(qtconfig, QLocale::c()).isValid())
         return std::make_unique<QSettings>(qtconfig, QSettings::IniFormat);
-#ifdef Q_OS_DARWIN
+/*#ifdef Q_OS_DARWIN
     CFBundleRef bundleRef = CFBundleGetMainBundle();
     if (bundleRef) {
         QCFType<CFURLRef> urlRef = CFBundleCopyResourceURL(bundleRef,
@@ -122,7 +122,7 @@ static std::unique_ptr<QSettings> findConfiguration()
                 return std::make_unique<QSettings>(qtconfig, QSettings::IniFormat);
         }
     }
-#endif
+#endif*/
     if (QCoreApplication::instanceExists()) {
         QString pwd = QCoreApplication::applicationDirPath();
         qtconfig = pwd + u"/qt" QT_STRINGIFY(QT_VERSION_MAJOR) ".conf"_s;
-- 
2.39.3 (Apple Git-146)

